contracts/SpokePool.sol

```diff
@@ -367,7 +388,7 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
         uint64 realizedLpFeePct,
         uint64 relayerFeePct,
         uint32 depositId
-    ) public nonReentrant {
+    ) public nonReentrant unpaused {
```

Comment:
Have we considered separate pause functionality for fills?

I think I can see a use case for pausing deposits but allowing fills to proceed. This would allow relayers to "drain" any remaining deposits that were made before the contract was paused, but which had not been filled in time. This would be especially useful in the case that we want to pause in order to perform administrative activities (as opposed to an emergency pause, in which case we may want to pause everything).

The reason I ask is that we really risk a messy situation if we end up with users funds stuck in a spoke pool with no way of obtaining a refund. I don't see any obvious way of making such users whole by paying them out-of-band and administratively nullifying the deposits, because then we'd need to manually orchestrate relayer refunds somehow.

Scenarios in which we might see "stuck" deposits include:
 - Users who have integrated the contracts directly.
 - Users of aggregators/other third-parties that have holes in their logic.

I don't anticipate this happening directly via the Across FE, since I presume we'd disable deposits in the FE a safe time before pausing the spoke pool.

Reply:
hmm this is a good point, I see no risk in having different pause functionality

Reply:
Marking as resolved, since this topic continues here: https://github.com/across-protocol/contracts-v2/pull/212